compiler-plugin/src/main/java/io/ballerina/stdlib/ai/plugin/GenerateMethodModificationTask.java

```diff
@@ -204,16 +251,14 @@ private class GenerateMethodJsonSchemaGenerator extends NodeVisitor {
         private static final String BYTE = "byte";
         private static final String NUMBER = "number";
         private final SemanticModel semanticModel;
-        private final Document document;
         private final TypeMapper typeMapper;
         private final TypeDefinitionSymbol modelProviderSymbol;
 
-        public GenerateMethodJsonSchemaGenerator(SemanticModel semanticModel, Document document,
+        public GenerateMethodJsonSchemaGenerator(SemanticModel semanticModel, Optional<ModuleSymbol> aiModuleSymbol,
                                                  AiCodeModifier.AnalysisData analyserData) {
             this.semanticModel = semanticModel;
-            this.document = document;
             this.typeMapper = analyserData.typeMapper;
-            this.modelProviderSymbol = getAiModelProviderSymbol(document.syntaxTree().rootNode()).orElse(null);
+            this.modelProviderSymbol = getAiModelProviderSymbol(aiModuleSymbol).orElse(null);
```

Comment:
Are you doing something here that cannot be done in L107? Why is it required to pass `aiModuleSymbol` all the way here and try extracting out the model provider symbol for every document instead of doing it once?